### PR TITLE
TECH-481: Create cucumber test for bb-identity POST /client-mgmt/oidc-client

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
Create cucumber test for POST /client-mgmt/oidc-client

Ticket: https://govstack-global.atlassian.net/browse/TECH-481

### CODE REVIEW

File: test/openAPI/features/client_create.feature
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/client_create.feature:31]:
```suggestion
    | 2011-12-01T14:48:00.000Z | health-service     | Health Service   | health-gov      | http://example4.com | idbb:acr:biometrics                | preferred_username | http://redirectMe4.com |
```

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]
1. I would also add a check if returned code matches with the reason why the error was returned.
I can see mockoon is prepared well for that, simple check is what is left to do.

2. Shouldn't the data be also sent in gherkin file? For now it is hardcoded.
image

3. I would recommend invoking invalid test cases before valid ones. I'm not sure but the response may come as `duplicate_client_id` if we first create a user with smoke test and then will try to invoke invalid test cases. Maybe let's make the order as smoke, negative, positive (negative would be sent with data different than smoke test)

**Comment by KarolinaKopacz** [comment]
@mgrochalskisoldevelo 
Ad.1. Added
Ad.2. There is too many data in this body. For the developers and also for the user it can be misleading to read all of this but only one parameter is invalid.

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [inline test/openAPI/features/client_create.feature:130]: Can we also add a scenario with empty string for redirectUris and expect invalid_redirect_uri ?

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [inline test/openAPI/features/client_create.feature:7]: redirectUris is missing to be mentioned here

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [inline test/openAPI/features/client_create.feature:130]: can we also add a scenrio for invalid_public_key when public key is not sent in a proper JWK format

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: test/openAPI/features/client_create.feature
Comment (KarolinaKopacz) [reply test/openAPI/features/client_create.feature:7]: It's missing because this is smoke test and `redirectUris` in the API spec is not reqiured

File: test/openAPI/features/client_create.feature
Comment (KarolinaKopacz) [reply test/openAPI/features/client_create.feature:13]: Good point!

File: test/openAPI/features/client_create.feature
Comment (KarolinaKopacz) [reply test/openAPI/features/client_create.feature:130]: In the API spec redirectUris is not required and can be just string so it means if I add `""` it should work. If redirectUris should have minimum 1 letter then this information has to be added do the API spec. For now there is no such information so software candidate will not know it.

File: test/openAPI/features/client_create.feature
Comment (KarolinaKopacz) [reply test/openAPI/features/client_create.feature:7]: Thanks. How this publicKey should looks before encode? I see it should be an object so what is a key it this object?

File: test/openAPI/features/client_create.feature
Comment (KarolinaKopacz) [reply test/openAPI/features/client_create.feature:130]: Of course!

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [reply test/openAPI/features/client_create.feature]: JWK formatted public key example is as below.
`{
            "kty": "RSA",
            "e": "AQAB",
            "use": "sig",
            "alg": "RS256",
            "n": "mykWIftknK1TQmbiazuik0rWGsxeOIUE3yfSQJgoCfdGXY4HfHE6AlNKFdIKZOXe-U-L21Klj692e9iZx05rHHaZvO0a4IzyFMOyw5wjBCWoBOcA4q93LPkZTSkIq9I2Vgr6Bzwu6X7QPMbmF8xAKX4KeSn_yZcsAhElHBOWkENmKp76yCyTeE4DAIGah1BcgiB_KWvOZOedwTRDLyQ0DZM1z07-N-rPh0qSd2UFRRY-b_jc9opjyRQq3d5ZkiB9W4ReAUhIKA9uc1RDs1shc3G8zgZp3qH6fYWmsOi23BOA_q8Z-wMHwPK2vEJvgZIWovAG5jGFbMilNcFQfzLJcQ"
        }`

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [reply test/openAPI/features/client_create.feature:132]: But I see the very first scenario, the client id is hardcoded as "e-health-service", am I following the test scenarios correctly ?

Comment (vishwa-vyom) [reply test/openAPI/features/client_create.feature:132]: ok, in the next iteration of the spec, we will add this

File: test/openAPI/features/client_create.feature
Comment (KarolinaKopacz) [reply test/openAPI/features/client_create.feature]: Thank you!

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [reply test/openAPI/features/client_create.feature:132]: I have created the below PR for making the redirectUris manditory and waiting for review from @jdub13 on this
https://github.com/GovStackWorkingGroup/bb-identity/pull/36

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [reply test/openAPI/features/client_create.feature]: Spoke to the internal team and found that redirectUris is a required field, so created the below PR on this and waiting for review.
#36

Comment (KarolinaKopacz) [reply test/openAPI/features/client_create.feature]: Ok, so I will also change the tests now and add redirectUri as required

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [inline test/openAPI/features/client_create.feature:44]: For this scenario the expected error code is invalid_request, as the new PR https://github.com/GovStackWorkingGroup/bb-identity/pull/36

File: test/openAPI/features/client_create.feature
Comment (vishwa-vyom) [reply test/openAPI/features/client_create.feature:132]: As per the scenarios, can we send the post request twice with same client id and verify whether it causes this issue ? That we don't hardcode any client id.

**Pull request merged**
